### DISCUSSION TITLE
请问代理设置有没有什么特殊的姿势

### DISCUSSION THREAD

**Discussion by xiaobaozi233** [Tsuk1ko/CQ-picfinder-robot#129] [问题求助] [answered]
我在服务器上设置了clash，然后代理http+https走clash的本地sock5端口。
在clash的日志中有看到向saucenao.com发起连接并走了代理通道。

但是还是都返回：
搜索失败惹 QAQ
有可能是服务器网络爆炸，请重试一次

想问问有没有用国内服务器的？各位的代理都怎么设置的，谢谢！

**Comment by Tsuk1ko** [comment]
看 `logs/error.log` 是什么报错

**Comment by xiaobaozi233** [comment]
1/19/2021, 3:08:07 PM [error] saucenao[0][request]
TypeError: Cannot read property 'message' of undefined
    at /root/cqpf/src/saucenao.js:143:30
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async doSearch (/root/cqpf/src/saucenao.js:45:3)
    at async searchImg (/root/cqpf/main.js:478:25)
{
  "message": "Cannot read property 'message' of undefined"
}

**Comment by Tsuk1ko** [comment]
搜图的时候加上 `--debug`，然后发一下 `logs/normal.log` 里输出的内容

**Comment by xiaobaozi233** [comment, marked as answer]
1/19/2021, 4:19:37 PM saucenao[0] saucenao.com
""
1/19/2021, 4:21:05 PM saucenao[0] saucenao.com
""

**Reply by Tsuk1ko** [reply]
🤔 看起来是代理的问题，回包竟是空的

**Reply by Tsuk1ko** [reply]
确定 `bot.proxy` 配置正确以及代理可以正常访问 saucenao.com 么

**Reply by xiaobaozi233** [reply]
我换个代理试试看

**Reply by xiaobaozi233** [reply]
我之前是走的全局代理有看到走代理访问，现在用bot.proxy写socks://ip:port和ip:port都没有走代理。是写法有问题吗？

**Reply by Tsuk1ko** [reply]
clash 我记得端口是 7890 还是 7891 来着，旧版 socks 和 http 分开，新版合并 7890，反正用 socks 的端口写 `socks://127.0.0.1:port` 即可

**Reply by xiaobaozi233** [reply]
我用clash和其他socks代理都没有成功，写法就是"proxy": "socks://ip:port"。查看代理日志并没有看到有通过代理发起链接。

**Reply by Tsuk1ko** [reply]
确定有重载配置或者重启本项目么

**Reply by xiaobaozi233** [reply]
--reload 和stop再start都试了一下，还是不行。匪夷所思

**Reply by xiaobaozi233** [reply]
仔细研究了一下怀疑是我在centos8上装的，有很多环境依赖有问题。。。
我今天手动装了一下pixman+cairo+pango+libjpeg库


现在搜索会返回一个结果，然后另外出现一条搜索失败惹 QAQ。也就是我发一个图会返回两条。
日志提示
1/20/2021, 5:21:08 PM [error] saucenao[0][request]
TypeError: Cannot read property 'message' of undefined
    at /root/cqpf/src/saucenao.js:143:30
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async doSearch (/root/cqpf/src/saucenao.js:45:3)
    at async searchImg (/root/cqpf/main.js:478:25)
{
  "message": "Cannot read property 'message' of undefined"
}

**Reply by Tsuk1ko** [reply]
重复响应的问题看 wiki 部署流程-错误排查-3
